### DISCUSSION TITLE
Get SqlContext from Request/Job

### DISCUSSION THREAD

**Discussion by kdesjard** [apalis-dev/apalis#271] [General]
How would I go about getting the SqlContext from a Request/Job?  It would be useful to be able to access the various db fields like the status (State), eg. Pending/Running/...

**Comment by geofmureithi** [comment]
These are stored in data/extensions.
You can do `req.data_opt::<SqlContext>()`.
See https://github.com/geofmureithi/apalis/blob/19aa3a2e1728ac39f4f8c6f4719879e39bc226aa/packages/apalis-sql/src/context.rs#L11
https://github.com/geofmureithi/apalis/blob/19aa3a2e1728ac39f4f8c6f4719879e39bc226aa/packages/apalis-sql/src/from_row.rs#L18

You can do:

```rs
async fn job(req: Email, ctx: SqlContext) {

}
```
The code above only works for Sql backed jobs

This is approach is not best, I think I will fix it in 0.6. Let me know if this helps.

**Reply by kdesjard** [reply]
data_opt doesn't exist anymore should this be replaced by get() ? 

Also, it doesn't seem like SqlContext is exported.

**Reply by geofmureithi** [reply]
You are correct. It should be exported. @AzHicham maybe you can combine this with the redis one. 
(Ps am on the road, thats why)
